<launch>
  <!-- Launch file for Sixi 2 robot arm with JETSON -->

  <!-- VIZUALIZATION NODE ______________________________________________________
  Purpose: Launch RVIZ -->
  <!-- TODO -->

  <!-- XBOX CONTROLLER NODE _____________________________________________________
  Purpose: Map the movement of the joystick to the desired cartesian movement (XYZ) of the robot -->
  <node pkg="sixi_robot" type="sixi_teleop_competition" name="arm_teleop"/>
  <node pkg="sixi_robot" type="inverse_kinematic.py" name="IK"/>

  <!-- HARDWARE INTERFACE NODE _________________________________________________
  Purpose: Communicate with Arduino microcontroller -->
  <node pkg="sixi_robot" type="hardware_interface.py" name="hardware_interface"/>
  <node pkg="rosserial_python" type="serial_node.py" name="rosserial">
    <param name="dev" type="string" value="/dev/ttyUSB0"/>
  </node>

  <!-- JOINT POSITION SERVER NODE ______________________________________________
  Purpose: Server to get the actual position of the robot -->
  <!--NOT WORKING WELL SO NOT USING FOR 2021 COMPETITION-->
  <!--<node pkg="joint_states_listener" type="joint_pos_server.py" name="joint_pos_server"/>-->

</launch>
